packages/flutter/lib/src/rendering/animated_size.dart

```diff
@@ -0,0 +1,144 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/animation.dart';
+import 'package:meta/meta.dart';
+
+import 'box.dart';
+import 'object.dart';
+import 'shifted_box.dart';
+
+/// A render object that can render child size change animations. It works by
+/// always animated from its old child's size to the new one with the given
+/// duration and curve. In case the child changes its size in every single frame
+/// the object deduces that its size is animated from somehwere else, it
+/// skips/stops its animation and immediately changes its size to match the
+/// child.
+///
+/// In case the child overflows the current animated size of the parent, it gets
+/// clipped automatically.
+class RenderAnimatedSize extends RenderAligningShiftedBox {
+  /// Creates a render animated size. [duration] and [curve] define the
+  /// animation. [alignment] is used to align the child in the case where the
+  /// parent is not (yet) the same size as the child. [onLayoutChanged] is
+  /// called every time the layout changes.
+  ///
+  /// [duration], [curve], and [alignment] should not be null.
+  RenderAnimatedSize({
+    Curve curve: Curves.linear,
+    RenderBox child,
+    FractionalOffset alignment: FractionalOffset.center,
+    @required Duration duration
+  }) : super(child: child, alignment: alignment) {
+    assert(duration != null);
+    assert(curve != null);
+    _controller = new AnimationController(
+      duration: duration
+    );
+    _animation = new CurvedAnimation(
+      parent: _controller,
+      curve: curve
+    );
+  }
+
+  AnimationController _controller;
+  CurvedAnimation _animation;
+  SizeTween _sizeTween = new SizeTween();
+  bool _didChangeTargetSizeLastFrame = false;
+  bool _hasVisualOverflow;
+
+  /// The duration of the animation.
+  Duration get duration => _controller.duration;
+  set duration(Duration value) {
+    assert(value != null);
+    if (value == _controller.duration)
+      return;
+    _controller.duration = value;
+  }
+
+  /// The curve of the animation.
+  Curve get curve => _animation.curve;
+  set curve(Curve value) {
+    assert(value != null);
+    if (value == _animation.curve)
+      return;
+    _animation.curve = value;
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    if (_animatedSize != _sizeTween.end)
+      _trigger();
+  }
+
+  @override
+  void detach() {
+    _controller?.dispose();
```

Comment:
detach() is not the same as dispose(). If you create the controller in the constructor you should never dispose it.

You should stop() it in detach(), and can restart it when you reattach, maybe?